mobilecoind/src/service.rs

```diff
@@ -559,11 +559,14 @@ impl<T: BlockchainConnection + UserTxConnection + 'static, FPR: FogPubkeyResolve
         let shared_secret =
             get_tx_out_shared_secret(account_key.view_private_key(), &tx_public_key);
 
-        let (value, _blinding) = tx_out
+        let (amount_data, _blinding) = tx_out
             .amount
             .get_value(&shared_secret)
             .map_err(|err| rpc_internal_error("amount.get_value", err, &self.logger))?;
 
+        // FIXME: We should not discard amount_data.token_id
```

Comment:
What should we do with it?

Reply:
I guess the `UnspentTxOut` object should store it

Reply:
we could store an `AmountData` instead of a value to try to keep the token id with the value